### PR TITLE
Variables and types section

### PR DESCRIPTION
**Pull request by jotas6**
Setup and content for the Julia variables and types section of the workshop.

## Status
- [x] Edit the PLACEHOLDER text
- [x] Edit README file
- [x] Code examples
- [x]  Instructor notes
- [x] Reference sheet
- [x] Schedule

### CODE REVIEW

File: docs/instructors.md [hunk 1]
```diff
@@ -1,10 +1,42 @@
...
+most users will use and come across `Float`s and `Integer`s. Go over the definition of `Integer`s, what they mean
+and how variables are assigned in Julia. Make sure to mention the use of underscores (`_`) as a useful tool to define 
+large numbers. Next, go over `Float`s following a similar approach to the one used to explain integers. Show 
+how you can create the same values as before, but as `Float`s (e.g, `1` and `1.0`) and use this opportunity to explain 
+the use of `typeof` to check a variable's type. Here you can show how to define numbers in scientific notation. Lastly, 
+go over all the math operations, mentioning the [PEMDAS order of operations](https://en.wikipedia.org/wiki/Order_of_operations),
+how to calculate square roots with `\sqrt<TAB>` and briefly showcasing rational and complex numbers.
+
+For `02-booleans.jl`, start by showing the possible comparisons between numbers (less than, greater than, etc.). After that,
+explain the use of `!` for negation, `||` for OR and `&&` for AND. Make sure to go over how these operators work if needed.
```
Comment (storopoli) [inline docs/instructors.md:24]: It's useful to add that some users might ask about single `|` and `?`. How the instructor would reply to that question?

File: docs/reference.md [hunk 2]
```diff
@@ -1,28 +1,80 @@
...
+| String interpolation | `$` | Used as `$(expression)` |
+| Pattern matching | `contains` and `occursin` | Used as `contains(haystack, needle)` and `occursin(needle, haystack)` |
+| Return a string with all capital letters | `uppercase` |  |
+| Return a string with all lower case letters | `lowercase` |  |
+| Return a string in title case | `titlecase` |  |
+| Replace a part of a string | `replace` | Used as `replace(str, original_part => new_part)` |
+| Reshape a `Vector` | `reshape` | Used as `reshape(vector, new_shape)`, where `new_shape` could be `(3, 3)` to create a 3x3 matrix (if `vector` contains the appropriate number of elements) |
+| Indexing or slicing | `vector[1]`, `vector[1:10]`, `matrix[1, 2]`, `matrix[1:3, 2:4]` | Indexing returns a single value, while slicing returns multiple values |
+ 
 ## Glossary
```
Comment (storopoli) [inline docs/reference.md:57]: We are missing `Float`, `Int` etc.

**Review by storopoli — CHANGES_REQUESTED** [review]

File: 01-variables/03-strings.jl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Join strings
+greeting = "Hello"
+name = "Jake"
+
+## using string()
+string(greeting, ", ", name)
+
+## using * (same as multiplication)
+greeting*", "*name # Watch out: + doesn't work
+
```
[the remark this answers is not in the document]
Comment (jotas6) [reply 01-variables/03-strings.jl:25]: Yes, I think that is much better

File: docs/reference.md (hunk 2 above)
Comment (jotas6) [reply docs/reference.md:57]: Would this be a description of the types? Something like this?

| Action | Command | Observations |
|-----------|-----------------|---------------------|
| Integer type | `Int` | |

**Comment by jotas6** [comment]
I think I have covered all the comments (except for the one where I left the question). I marked the conversations are resolved to keep track of what I was doing, but I could make more adjustments if the solutions are not good.

File: docs/reference.md (hunk 2 above)
Comment (storopoli) [reply docs/reference.md:60]: Yes that would be good.

**Review by storopoli — APPROVED** [review]

**Pull request merged**